data_model/src/lib.rs

```diff
@@ -397,6 +398,24 @@ impl IdentifiableBox {
     }
 }
 
+impl<'live> TryFrom<&'live IdentifiableBox> for Box<&'live dyn HasMetadata> {
+    type Error = ();
+
+    fn try_from(
+        v: &'live IdentifiableBox,
+    ) -> Result<Box<&'live (dyn HasMetadata + 'live)>, Self::Error> {
+        match v {
+            IdentifiableBox::NewDomain(v) => Ok(Box::new(v.as_ref())),
+            IdentifiableBox::NewAccount(v) => Ok(Box::new(v.as_ref())),
+            IdentifiableBox::NewAssetDefinition(v) => Ok(Box::new(v.as_ref())),
+            IdentifiableBox::Domain(v) => Ok(Box::new(v.as_ref())),
+            IdentifiableBox::Account(v) => Ok(Box::new(v.as_ref())),
+            IdentifiableBox::AssetDefinition(v) => Ok(Box::new(v.as_ref())),
+            _ => Err(()),
+        }
+    }
+}
```

Comment:
```suggestion
impl<'live> TryFrom<&'live IdentifiableBox> for &'live dyn HasMetadata {
    type Error = ();

    fn try_from(
        v: &'live IdentifiableBox,
    ) -> Result<&'live (dyn HasMetadata + 'live), Self::Error> {
        match v {
            IdentifiableBox::NewDomain(v) => Ok(v.as_ref()),
            IdentifiableBox::NewAccount(v) => Ok(v.as_ref()),
            IdentifiableBox::NewAssetDefinition(v) => Ok(v.as_ref()),
            IdentifiableBox::Domain(v) => Ok(v.as_ref()),
            IdentifiableBox::Account(v) => Ok(v.as_ref()),
            IdentifiableBox::AssetDefinition(v) => Ok(v.as_ref()),
            _ => Err(()),
        }
    }
}
```

this should be sufficient